src/cata_tiles.cpp

```diff
@@ -887,6 +888,90 @@ void cata_tiles::draw( int destx, int desty, const tripoint &center, int width,
     SDL_RenderSetClipRect(renderer, NULL);
 }
 
+void cata_tiles::draw_minimap( int destx, int desty, const tripoint &center, int width, int height )
+{
+    if (!g) {
+        return;
+    }
+
+    int start_x = center.x - 55;
+    int start_y = center.y - 55;
+    int tiles_x = 110;
+    int tiles_y = 110;
+    int tile_size_x = std::max(width / tiles_x, 1);
+    int tile_size_y = std::max(height / tiles_y, 1);
+
+    // Center the drawn area within the total area.
+    int drawn_width = tiles_x * tile_size_x;
+    int drawn_height = tiles_y * tile_size_y;
+    int border_width = (width - drawn_width) / 2;
+    int border_height = (height - drawn_height) / 2;
+
+    auto &ch = g->m.access_cache( center.z );
+
+    // First draw terrain.
+    for( int y = 0; y < tiles_y; y++) {
+        for( int x = 0; x <= tiles_x; x++) {
+            tripoint p(start_x + x, start_y + y, center.z);
+
+            lit_level lighting = ch.visibility_cache[p.x][p.y];
+            SDL_Color color;
+            if(lighting == LL_DARK || lighting == LL_BLANK) {
+                color.r = 12;
+                color.g = 12;
+                color.b = 12;
+            } else {
+                int veh_part = 0;
+                vehicle *veh = g->m.veh_at( p, veh_part );
+                auto furniture = g->m.furn_at( p );
+                if (veh != nullptr) {
+                    color = cursesColorToSDL(veh->part_color(veh_part));
+                } else if ( g->m.has_furn( p ) ) {
+                    auto furniture = g->m.furn_at( p );
```

Comment:
This (and `terrain` below) should be references (`furn_at` returns references, theres is no need to copy them).